### PR TITLE
[INTERNAL] Enhance Dependabot configuration to cover also v2 branch

### PR DESCRIPTION
**Pull request by flovogt**

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,13 +4,32 @@ updates:
   directory: "/"
   schedule:
     interval: "weekly"
+  target-branch: "main"
```
Comment (matz3) [inline .github/dependabot.yml:7]: If I understand the docs correctly, target-branch should not be provided for the default branch as this prevents the settings to be used for security updates: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#target-branch

**Review by matz3 — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (flovogt) [reply .github/dependabot.yml:7]: > Use target-branch to specify a different branch for manifest files and for pull requests. When you use this option, the settings for this package manager will no longer affect any pull requests raised for security updates.

Does this mean in general no security updates?

Comment (matz3) [reply .github/dependabot.yml]: I think it just means the settings are not considered for them.

**Review by matz3 — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (flovogt) [reply .github/dependabot.yml]: `target-branch` is removed from the main branch config

**Pull request merged**